workspace/apps/pidp/src/app/features/auth/resolvers/bc-provider-completed.resolver.ts

```diff
@@ -0,0 +1,30 @@
+import { inject } from '@angular/core';
+import { ResolveFn, Router } from '@angular/router';
+import { PartyService } from '@app/core/party/party.service';
+import { AccessRoutes } from '@app/features/access/access.routes';
+import { StatusCode } from '@app/features/portal/enums/status-code.enum';
+import { ProfileStatus } from '@app/features/portal/models/profile-status.model';
+import { PortalResource } from '@app/features/portal/portal-resource.service';
+import { map } from 'rxjs';
+
+export const bcProviderCompletedResolver: ResolveFn<
+  boolean | null | Promise<boolean>
+> = () => {
+  const partyId = inject(PartyService).partyId;
+  const portalResource = inject(PortalResource);
+  const router = inject(Router);
+  return portalResource.getProfileStatus(partyId).pipe(
+    map((profileStatus: ProfileStatus | null) => {
+      if (!profileStatus) {
+        return null;
+      }
+
+      return (
+        profileStatus.status.bcProvider.statusCode !== StatusCode.COMPLETED ||
```

Comment:
When I read the backend code, the status code could be:
- 1 AVAILABLE
- 2 COMPLETED
- 3 NOT_AVAILABLE
- 5 HIDDEN

But with this condition, the user can access the route even if the status is HIDDEN or NOT_AVAILABLE